### PR TITLE
refactor(calibration_storage): remove top level if statement for calibration storage exports

### PR DESCRIPTION
**Pull request by Laura-Danielle**
# Overview

In the interest of getting auto-calibration out, there were a few corners cut in the calibration storage refactor. The first major one was how exports were handled. This PR addresses that issue by checking the robot feature flag inside functions defined in the `__init__.py` file of `calibration_storage`.

# Test Plan

- [ ] Go through all of the OT-2 calibration flows and ensure that data is still saving
- [ ] Go through auto-calibration for a single channel pipette and gripper to make sure the data is still saving
- [ ] upgrade/downgrade from this branch to make sure nothing weird happens

# Changelog

- Define all of the calibration storage functions inside the `__init__.py` of `calibration_storage` and check the feature flag at function call time
- Modify code base to correctly import these new functions
- Modify `calibration_storage` tests to mock out the file operator functions

# Review requests

Let me know if this implementation seems OK. The next PR after this will address combining the instrument calibration files in the hardware controller into one file with an `InstrumentCalibrationHandler` class.

# Risk assessment

Medium. Affects calibration flows on the ot2.

### CODE REVIEW

File: api/src/opentrons/calibration_storage/ot2/tip_length.py [hunk 1]
```diff
@@ -227,4 +217,5 @@ def save_tip_length_calibration(
     dict_of_tip_lengths = {}
     for key, item in all_tip_lengths.items():
         dict_of_tip_lengths[key] = json.loads(item.json())
+    print("save being called!")
```
Comment (sfoster1) [inline api/src/opentrons/calibration_storage/ot2/tip_length.py:220]:
```suggestion

```

File: api/src/opentrons/calibration_storage/ot3/__init__.py [hunk 2]
```diff
@@ -2,9 +2,41 @@
...
+    get_robot_deck_attitude as ot3_get_robot_deck_attitude,
+    delete_robot_deck_attitude as ot3_delete_robot_deck_attitude,
+)
+from .pipette_offset import (
+    save_pipette_calibration as ot3_save_pipette_calibration,
+    get_pipette_offset as ot3_get_pipette_offset,
+    clear_pipette_offset_calibrations as ot3_clear_pipette_offset_calibrations,
+    delete_pipette_offset_file as ot3_delete_pipette_offset_file,
+)
+from .tip_length import (
```
Comment (sfoster1) [inline api/src/opentrons/calibration_storage/ot3/__init__.py:16]: all of these functions are in a submodule called `ot3`, I don't think they need the `ot3_` prefix in their names. if a caller needs to have them alongside the ot2 ones and differentiate them, they can just import the `ot3` submodule and call `ot3.save_tip_length_calibration` or `ot2.save_tip_length_calibration`

File: api/src/opentrons/calibration_storage/__init__.py [hunk 3]
```diff
@@ -13,48 +20,239 @@
...
+    ot3_models.v1.TipLengthModel,
+]
+
+CalibrationStatusType = typing.Union[
+    ot2_models.v1.CalibrationStatus,
+    ot3_models.v1.CalibrationStatus,
+]
+
+
+def save_robot_deck_attitude(
```
Comment (sfoster1) [inline api/src/opentrons/calibration_storage/__init__.py:47]: I think we should be reducing our use of the ot3/ot2 feature flag. these changes add a whole lot of changes, which kinda magically change behavior... I'd rather have the decision about "which featureset to use" moved as high as possible, and then have the usage of these functions be done based on function arguments or dependency injection or something.

**Review by sfoster1 — CHANGES_REQUESTED** [review]
To expand on an inline comment, I think we should lessen our use of the ot3 feature flag in general, and in particular try to avoid checking it to alter the functionality of free functions like this. it's hard to understand the scope and effects of a feature flag like this when it's referenced all over the place, toward the leaves of the program tree - you can never be sure, when you look at a function call, whether it's going to have its effects swapped out from under it by the feature flag.

instead, I think we should be moving towards having our software accept what's given to it, I guess. like, check the feature flag just in [`opentrons.initialize`](https://github.com/Opentrons/opentrons/blob/8d2b1c85ca034a52c6f7150b9cb51b584121782f/api/src/opentrons/__init__.py#L138) and have everything else deal with its effects. maybe there's a query on the type of hardware controller that's returned, maybe the hardware controller owns and exposes a reference to its calibration storage interface on its own terms... just less feature flag checking.

**Comment by b-cooper** [comment]
Closing this PR for now, but the branch is sticking around

**Pull request closed without merging**